builtin/providers/softlayer/resource_softlayer_user_customer.go

```diff
@@ -0,0 +1,400 @@
+	return &schema.Resource{
+		Create: resourceSoftLayerUserCustomerCreate,
+		Read:   resourceSoftLayerUserCustomerRead,
+		Update: resourceSoftLayerUserCustomerUpdate,
+		Delete: resourceSoftLayerUserCustomerDelete,
+		Exists: resourceSoftLayerUserCustomerExists,
+
+		Schema: map[string]*schema.Schema{
+			"id": &schema.Schema{
+				Type:     schema.TypeInt,
+				Computed: true,
+			},
+
+			"username": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"first_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"last_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"email": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"company_name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"address1": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"address2": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"city": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"state": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"country": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			// TODO Support more intuitive string values for timezone and user_status
+			"timezone": &schema.Schema{
+				Type:     schema.TypeInt,
+				Required: true,
+			},
+			"user_status": &schema.Schema{
+				Type:     schema.TypeInt,
+				Optional: true,
+				// Active by default
+				Default: 1001,
+			},
+			"password": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				StateFunc: func(v interface{}) string {
+					hash := sha1.Sum([]byte(v.(string)))
+					return hex.EncodeToString(hash[:])
+				},
+			},
+			"permissions": &schema.Schema{
+				Type:     schema.TypeSet,
+				Optional: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Set:      schema.HashString,
+			},
+			"has_api_key": &schema.Schema{
+				Type:     schema.TypeBool,
+				Optional: true,
+				Default:  false,
+			},
+			"api_key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+		},
+	}
+}
+
+// Create a SoftLayer_User_Customer_CustomerPermission_Permission object from the given string input
+func makePermission(p string) datatypes.SoftLayer_User_Customer_CustomerPermission_Permission {
+	return datatypes.SoftLayer_User_Customer_CustomerPermission_Permission{
+		Key:     "",
+		Name:    "",
+		KeyName: p,
+	}
+}
+
+// Convert a "set" of permission strings to a list of SoftLayer_User_Customer_CustomerPermission_Permissions
+func getPermissions(d *schema.ResourceData) []datatypes.SoftLayer_User_Customer_CustomerPermission_Permission {
+	permissionsSet := d.Get("permissions").(*schema.Set)
+
+	if permissionsSet.Len() == 0 {
+		return nil
+	} else {
```

Comment:
From golint: `if block ends with a return statement, so drop this else and outdent its block`